app/models/song.rb

```diff
@@ -16,4 +18,8 @@ def to_param
   def generate_slug
     self.slug = title.parameterize
   end
+
+  def self.find_by_rating(song)
+    Song.where.not(title: song.title).where(rating: song.rating).limit(3)
```

Comment:
it's more efficient to do `where.not(id: song.id)` than looking at the title since integer lookups (plus primary keys) are WAY faster for the database than strings